ribbon-httpclient/src/main/java/com/netflix/niws/client/http/RestClient.java

```diff
@@ -155,10 +156,10 @@ public void initWithNiwsConfig(IClientConfig clientConfig) {
         this.config = new DefaultApacheHttpClient4Config();
         this.config.getProperties().put(
                 ApacheHttpClient4Config.PROPERTY_CONNECT_TIMEOUT,
-                Integer.parseInt(String.valueOf(ncc.getProperty(CommonClientConfigKey.ConnectTimeout))));
+                ncc.getOrDefault(CommonClientConfigKey.ConnectTimeout));
```

Comment:
getOrDefault would return the value or a 0 here? Previously, if getProperty returned null, wouldn't this all explode in some way? I'm worried we may be returning zeroes where we used to throw an exception.

Reply:
This would have likely blown up before.  I'm OK with switching this to using Optionals.  Either way, I don't think this specific class is used internally at Netflix.